control-plane/rest/service/src/v0/nodes.rs

```diff
@@ -56,6 +58,35 @@ impl apis::actix_server::Nodes for RestApi {
         let node = client().drain(id.into(), label).await?;
         Ok(node.into())
     }
+
+    async fn put_node_label(
+        Path((id, label)): Path<(String, String)>,
+        Query(overwrite): Query<Option<bool>>,
+    ) -> Result<models::Node, RestError<RestJsonError>> {
+        let mut label_map = HashMap::new();
+        let mut iter = label.split('=');
+        let overwrite = overwrite.unwrap_or(false);
+
+        if let (Some(key), Some(value)) = (iter.next(), iter.next()) {
+            label_map.insert(key.trim().to_string(), value.trim().to_string());
+        } else {
+            return Err(RestError::from(ReplyError {
```

Comment:
Why don't we just take key/val from openapi?

Reply:
Because all the validation are done in control plane. If I pass hashmap then the validation will be done on extensions